### DISCUSSION TITLE
How to create a sqlalchemy Table with sql query on the fly?

### DISCUSSION THREAD

**Discussion by tonyabracadabra** [sqlalchemy/sqlalchemy#8211] [Usage Questions]
I am trying to create a sqlalchemy Table from sql query where I can get its columns via `table.c`, how am I suppose to do it? I am looking for something similar to the thing below.

```python
from sqlalchemy import Table
from sqlalchemy.sql import text

# creating a sqla Table object from sql query from text 
# on the fly WITHOUT creating the real table in the database

table = Table(text("SELECT * FROM table1"))
col = table.c.fieldname
```

**Comment by CaselIT** [comment]
Hi,

That's not something that's supported.
You can manually define columns in a text using [`columns()`](https://docs.sqlalchemy.org/en/14/core/sqlelement.html?highlight=text#sqlalchemy.sql.expression.TextClause.columns)
Depending on what you need to do you may also define a lowecase [`table`](https://docs.sqlalchemy.org/en/14/core/selectable.html#sqlalchemy.sql.expression.table), or reflect the Table (in this case `table1`) with [autoload_with](https://docs.sqlalchemy.org/en/14/core/metadata.html?highlight=table#sqlalchemy.schema.Table.params.autoload_with) or create a view from the query in the db and reflect that view as a Table

**Reply by tonyabracadabra** [reply]
thanks!!

**Comment by gordthompson** [comment]
You could achieve something similar to a `Table` using a CTE

```python
table1 = sa.Table("table1", sa.MetaData(), autoload_with=engine)
cte = sa.select(table1).where(table1.c.id < 2).cte(name="my_cte")
col = cte.c.fieldname
print(col)  # my_cte.fieldname
```

or a subquery

```python
table1 = sa.Table("table1", sa.MetaData(), autoload_with=engine)
subq = sa.select(table1).where(table1.c.id < 2).subquery(name="my_subquery")
col = subq.c.fieldname
print(col)  # my_subquery.fieldname
```

**Reply by tonyabracadabra** [reply]
Thank you! This is exactly what I want

**Reply by tonyabracadabra** [reply]
For a further question, can I have something like

```python
cte_table = text(sql).cte("tablename")
col = cte_table.c.fieldname
```

since my ultimate goal was to create such table from plain sql query in text
